rs/orchestrator/src/firewall.rs

```diff
@@ -1028,6 +1144,23 @@ mod tests {
             .expect("Failed to add node record.");
     }
 
+    /// Adds an [`ApiBoundaryNodeRecord`] to the registry.
+    fn add_api_boundary_node_record(
+        registry_data_provider: &Arc<ProtoRegistryDataProvider>,
```

Comment:
It's recommended to take direct references unless `Arc` features are necessary - it looks like `.add` takes a reference and no methods of `Arc` is invoked. By taking a reference, the code will also work with other types of smart pointers in the future and there is no disadvantage.

Reply:
Thanks for the suggestion! I didn't think much and copied 🫣